src/Jaeger/Util/Utils.cs

```diff
@@ -43,7 +43,10 @@ public static long UniqueId()
             while (value == 0)
             {
                 var bytes = new byte[8];
-                Random.NextBytes(bytes);
+                lock (Random)
```

Comment:
That is another option but it does make the change more complex. Need to initialize the multiple Random instances in a way that they all start with different seeds, otherwise a group of them could all output the same sequence if created near the same time. There are ways to address this, but it's more than a one line change (.NET Core handles this already, but I don't know that the .NET Framework does at this time).
Also, that would allocate a separate System.Random instance per thread that gets encountered. Even if there would never be contention, there would still be multiple Random instances created. Not a huge deal, but it is more instances that may not even be needed if there isn't highly concurrent access.
I would prefer to go with an ObjectPool if there's concern about the lock contention, but I feel that either approach is unnecessary complexity to optimize for a scenario that is likely far outside the any normal use case.

If a larger change is desired for this then I'm happy to explore the impact of either a ThreadLocal or ObjectPool implementation.

Reply:
probably fine to go with the lock unless there's actually a performance concern. I assume the Span is being locked for every write already anyway, so this change will add a relatively small amt of contention.